src/microbe_stage/MicrobeSpawner.cs

```diff
@@ -179,85 +156,29 @@ public static PackedScene LoadMicrobeScene()
         return GD.Load<PackedScene>("res://src/microbe_stage/Microbe.tscn");
     }
 
-    public static FloatingChunk SpawnChunk(ChunkConfiguration chunkType,
-        Vector3 location, Node worldNode, PackedScene chunkScene,
-        CompoundCloudSystem cloudSystem, Random random)
+    public void SetCurrentGame(GameProperties currentGame)
     {
-        var chunk = (FloatingChunk)chunkScene.Instance();
-
-        // Settings need to be applied before adding it to the scene
-        var selectedMesh = chunkType.Meshes.Random(random);
-        chunk.GraphicsScene = selectedMesh.LoadedScene;
-        chunk.ConvexPhysicsMesh = selectedMesh.LoadedConvexShape;
-
-        if (chunk.GraphicsScene == null)
-            throw new ArgumentException("couldn't find a graphics scene for a chunk");
-
-        // Pass on the chunk data
-        chunk.Init(chunkType, cloudSystem, selectedMesh.SceneModelPath);
-        chunk.UsesDespawnTimer = !chunkType.Dissolves;
-
-        worldNode.AddChild(chunk);
-
-        // Chunk is spawned with random rotation
-        chunk.Transform = new Transform(new Quat(
-                new Vector3(0, 1, 1).Normalized(), 2 * Mathf.Pi * (float)random.NextDouble()),
-            location);
-
-        chunk.GetNode<Spatial>("NodeToScale").Scale = new Vector3(chunkType.ChunkScale, chunkType.ChunkScale,
-            chunkType.ChunkScale);
-
-        chunk.AddToGroup(Constants.FLUID_EFFECT_GROUP);
-        chunk.AddToGroup(Constants.AI_TAG_CHUNK);
-        return chunk;
-    }
-
-    public static PackedScene LoadChunkScene()
-    {
-        return GD.Load<PackedScene>("res://src/microbe_stage/FloatingChunk.tscn");
-    }
-
-    public static void SpawnCloud(CompoundCloudSystem clouds, Vector3 location,
-        Compound compound, float amount)
-    {
-        int resolution = Settings.Instance.CloudResolution;
-
-        // This spreads out the cloud spawn a bit
-        clouds.AddCloud(compound, amount, location + new Vector3(0 + resolution, 0, 0));
-        clouds.AddCloud(compound, amount, location + new Vector3(0 - resolution, 0, 0));
-        clouds.AddCloud(compound, amount, location + new Vector3(0, 0, 0 + resolution));
-        clouds.AddCloud(compound, amount, location + new Vector3(0, 0, 0 - resolution));
-        clouds.AddCloud(compound, amount, location + new Vector3(0, 0, 0));
+        this.currentGame = currentGame;
     }
 
-    /// <summary>
-    ///   Spawns an agent projectile
-    /// </summary>
-    public static AgentProjectile SpawnAgent(AgentProperties properties, float amount,
-        float lifetime, Vector3 location, Vector3 direction,
-        Node worldRoot, PackedScene agentScene, Node emitter)
+    public List<ISpawned> Spawn(Node worldNode, Vector3 location, MicrobeSpecies species, bool isWanderer)
```

Comment:
This now creates the microbes all at once. The old system used enumerables to make sure that only the microbe instances that were to be spawned in that frame actually are spawned. I much prefer that earlier design as this could have problems regarding leaving undeleted microbes if the game is loaded / exited to menu while they are Microbe objects that have yet to be attached to the game scene.